src/Pirouette/SMT/FromTerm.hs

```diff
@@ -125,17 +104,14 @@ translateTerm knownNames (Raw.App var args) = case var of
       Just t -> return t
   Raw.Free (TermSig name) -> do
     _ <- traceMe ("translateApp: " ++ show name) (return ())
-    defn <- liftTranslator ("translateApp: Unknown name: " <> show name) (prtDefOf name)
+    defn <- lift $ lift $ prtDefOf name
```

Comment:
Very curious `lift $ lift` is not equivalent to `lift` in this context, but the type error is not too obvious at a first glance.

Reply:
Maybe this is the monomorphic `lift` which only "goes up one step"?

Reply:
I think it's because the monad in question is a `ReaderT x (ReaderT y)`, so lift gets confused